File: PlacementCellBackend/Controllers/JobsandInterships/ExperienceOpeningsController.cs
using Microsoft.AspNetCore.Mvc;
using PlacementCellBackend.DTOs.AlumniJobOpenings;
using PlacementCellBackend.Services.Placements.Interfaces;

namespace PlacementCellBackend.Controllers.Alumni
{
    [ApiController]
    [Route("api/[controller]")]
    public class ExperienceOpeningsController : ControllerBase
    {
        private readonly IAlumniPlacementService _alumniPlacementService;

        public ExperienceOpeningsController(IAlumniPlacementService alumniPlacementService)
        {
            _alumniPlacementService = alumniPlacementService;
        }

        [HttpGet]
        public async Task<ActionResult<IEnumerable<AlumniJobOpeningDto>>> GetAllAlumniJobPostion()
        {
            var placements = await _alumniPlacementService.GetAllAlumniJobPostionAsync();
            return Ok(placements);
        }

        [HttpGet("{id}")]
        public async Task<ActionResult<AlumniJobOpeningDto>> GetAlumniPlacementById(int id)
        {
            var placement = await _alumniPlacementService.GetAlumniPlacementByIdAsync(id);
            if (placement == null)
                return NotFound();
            return Ok(placement);
        }

        [HttpPost]
        public async Task<bool> CreateAlumniPlacement(AlumniJobOpeningCreateDto alumniPlacement)
        {
            var created = await _alumniPlacementService.CreateAlumniPlacementAsync(alumniPlacement);
            return created;
        }

        [HttpPut("{id}")]
        public async Task<IActionResult> UpdateAlumniPlacement(int id, AlumniJobOpeningCreateDto updatedAlumniPlacement)
        {

            var success = await _alumniPlacementService.UpdateAlumniPlacementAsync(id, updatedAlumniPlacement);
            if (!success)
                return NotFound();

            return NoContent();
        }

        [HttpDelete("{id}")]
        public async Task<IActionResult> DeleteAlumniPlacement(int id)
        {
            var success = await _alumniPlacementService.DeleteAlumniPlacementAsync(id);
            if (!success)
                return NotFound();

            return NoContent();
        }
    }
}

File: PlacementicsBackend/src/main/java/com/example/placementicsbackend/controllers/JobsandInterships/ExperienceOpeningsController.java
package com.example.placementicsbackend.controllers.JobsandInterships;

import com.example.placementicsbackend.dtos.AlumniJobOpenings.AlumniJobOpeningCreateDto;
import com.example.placementicsbackend.dtos.AlumniJobOpenings.AlumniJobOpeningDto;
import com.example.placementicsbackend.services.Placements.Interfaces.IAlumniPlacementService;
import lombok.RequiredArgsConstructor;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@RestController
@RequestMapping("/api/experience-openings")
@RequiredArgsConstructor
public class ExperienceOpeningsController {

    private final IAlumniPlacementService alumniPlacementService;

    @GetMapping
    public ResponseEntity<List<AlumniJobOpeningDto>> getAllAlumniJobPositions() {
        return ResponseEntity.ok(alumniPlacementService.getAllAlumniJobPositions());
    }

    @GetMapping("/{id}")
    public ResponseEntity<AlumniJobOpeningDto> getAlumniPlacementById(@PathVariable Integer id) {
        AlumniJobOpeningDto placement = alumniPlacementService.getAlumniPlacementById(id);
        if (placement == null) {
            return ResponseEntity.notFound().build();
        }
        return ResponseEntity.ok(placement);
    }

    @PostMapping
    public ResponseEntity<Boolean> createAlumniPlacement(@RequestBody AlumniJobOpeningCreateDto alumniPlacement) {
        boolean created = alumniPlacementService.createAlumniPlacement(alumniPlacement);
        if (!created) {
            return ResponseEntity.status(HttpStatus.BAD_REQUEST).body(false);
        }
        return ResponseEntity.status(HttpStatus.CREATED).body(true);
    }

    @PutMapping("/{id}")
    public ResponseEntity<Void> updateAlumniPlacement(
            @PathVariable Integer id,
            @RequestBody AlumniJobOpeningCreateDto updatedAlumniPlacement) {
        if (!alumniPlacementService.updateAlumniPlacement(id, updatedAlumniPlacement)) {
            return ResponseEntity.notFound().build();
        }
        return ResponseEntity.noContent().build();
    }

    @DeleteMapping("/{id}")
    public ResponseEntity<Void> deleteAlumniPlacement(@PathVariable Integer id) {
        if (!alumniPlacementService.deleteAlumniPlacement(id)) {
            return ResponseEntity.notFound().build();
        }
        return ResponseEntity.noContent().build();
    }
}
